code-challenges/object-keys/challenges-06.test.js

```diff
@@ -0,0 +1,247 @@
+  finalExam: true };
+
+const getCourseKeys = (obj) => {
+  // Solution code here...
+  return Object.keys(obj);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+Use the characters data below for the remainder of the challenges.
+
+Write a function named getHouses that returns a new array containing the names of all of the houses in the data set.
+------------------------------------------------------------------------------------------------ */
+
+let characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon A.',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn'
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister'
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen'
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell'
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    children: [],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon S.',
+    spouse: null,
+    children: [],
+    house: 'Snow'
+  }
+];
+
+const getHouses = (arr) => {
+  let houses = [];
+  for(let i=0; i<arr.length; i++){
+    houses.push(arr[i].house);
+  }
+  // Solution code here...
+  return houses;
+};
+
+/*------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named hasChildrenValues that uses Object.values to determine if any given character in the data set has children.
+
+This function should take in an array of data and a character name and return a Boolean.
+
+For example:
+hasChildrenValues(characters, 'Cersei') will return true
+hasChildrenValues(characters, 'Eddard') will return false
+------------------------------------------------------------------------------------------------ */
+
+const hasChildrenValues = (arr, character) => {
+  // Solution code here..
+  for(let i =0; i<arr.length; i++){
+    if(arr[i].name === character){
+      if((arr[i].children).length === 0){
+        return false;
+      }else{
+        return true;
+      }
+    }
+  }
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named hasChildrenEntries that is similar to your hasChildrenValues function from challenge 3, but uses the data's entries instead of its values.
+
+The input and output of this function are the same as the input and output from challenge 3.
+------------------------------------------------------------------------------------------------ */
+
+const hasChildrenEntries = (arr, character) => {
+  // Solution code here...
+  let entries = arr.entries();
+  for(let i of entries){
```

Comment:
I worry that `i` is a confusing variable name. `i` typically stands for index, a number, keeps track of how many times a loop has looped